podman/entrypoint.bash

```diff
@@ -0,0 +1,138 @@
+cp /dataverse/target/dataverse-*.war /jacoco-tmp/dataverse.war
+cd /jacoco-tmp && jar xf dataverse.war && rm dataverse.war
+mv /jacoco-tmp/WEB-INF/classes /jacoco-tmp/WEB-INF/classes-orig
+cd /jacoco-tmp && java -jar /jacoco/lib/jacococli.jar instrument WEB-INF/classes-orig/ --dest WEB-INF/classes
+cd /jacoco-tmp && jar cf /jacoco-tmp/dataverse.war *
+echo ""
+
+echo "building dvinstall.zip"
+cd /dataverse/scripts/installer && make clean && make dvinstall.zip
+unzip -q /dataverse/scripts/installer/dvinstall.zip -d /
+# allow unprivileged installer to write to /dvinstall
+/bin/chmod o+w /dvinstall
+echo "copying instrumented warfile into /dvinstall"
+cp /jacoco-tmp/dataverse.war /dvinstall/dataverse.war
+echo ""
+
+echo "starting postgres"
+sudo -u postgres /usr/pgsql-13/bin/pg_ctl start -D /var/lib/pgsql/data &
+/bin/sleep 1
+echo ""
+
+# start solr
+# current incompatibility in centos7 with sudo and raised values in /etc/security/limits.conf
+# just run solr as root for now instead. sigh.
+#sudo -u solr /usr/local/solr/bin/solr start
+cp /dataverse/conf/solr/8.8.1/*.xml /usr/local/solr/server/solr/collection1/conf/
+echo "starting solr"
+/usr/local/solr/bin/solr start -force
+
+# start payara
+echo "launching payara..."
+sudo -u payara /usr/local/payara5/bin/asadmin start-domain
+echo ""
+
+# deploy
+echo "deploying dataverse. stdout and stderr in /dvinstall/install.out."
+cd /dvinstall && sudo -u payara python3 install.py -f --config_file=default.config --noninteractive > install.out 2>&1
+echo "setting FAKE DOI Provider"
+curl -s -X PUT -d FAKE http://localhost:8080/api/admin/settings/:DoiProvider
+echo ""
+
+# restart payara
+echo "restarting payara..."
+sudo -u payara /usr/local/payara5/bin/asadmin stop-domain
+sudo -u payara /usr/local/payara5/bin/asadmin start-domain
+echo ""
+
+echo "preparing Dataverse to run integration tests"
+curl -s -X PUT -d burrito http://localhost:8080/api/admin/settings/BuiltinUsers.KEY
+export API_TOKEN=`cat /dvinstall/setup-all.*.log |grep apiToken | jq .data.apiToken |tr -d \"`
+curl -H "X-Dataverse-key: $API_TOKEN" --header "Content-Type: application/json" --request POST \
+  --data '{"assignee":":authenticated-users","role":"fullContributor"}' \
+  http://localhost:8080/api/dataverses/root/assignments
+curl -H "X-Dataverse-key: $API_TOKEN" -X POST http://localhost:8080/api/dataverses/root/actions/:publish
+# integration tests require sequential identifiers
+curl -O https://raw.githubusercontent.com/IQSS/dataverse/develop/doc/sphinx-guides/source/_static/util/createsequence.sql
+sudo -u postgres psql -d dvndb -f createsequence.sql
+echo "running integration tests"
+cd /dataverse && source /etc/profile.d/maven.sh && \
+   mvn test -Dtest=DataversesIT,DatasetsIT,SwordIT,AdminIT,BuiltinUsersIT,UsersIT,UtilIT,ConfirmEmailIT,FileMetadataIT,FilesIT,SearchIT,InReviewWorkflowIT,HarvestingServerIT,MoveIT,MakeDataCountApiIT,FileTypeDetectionIT,EditDDIIT,ExternalToolsIT,AccessIT,DuplicateFilesIT,DownloadFilesIT,LinkIT
```

Comment:
Instead of hard coding the list on integration tests here, can we call run-test-suite.sh? This thing:

https://github.com/IQSS/dataverse/blob/develop/conf/docker-aio/run-test-suite.sh

Reply:
I'm with you, but there's a small chicken-and-egg thing going on (concerning the location of the maven binary).

Once we're running integration tests in podman rather than ec2, the maven bit can come out of `run-test-suite.sh` which will make Pkiraly happy. Once run-test-suite.sh doesn't source that explicitly, the script will use the podman-branch-specified /etc/profile.d/maven.sh

So... eventually?

Reply:
I see. One thought is to treat the list of tests as data. That is, add a flat file that only has...

`DataversesIT,DatasetsIT,SwordIT,AdminIT,BuiltinUsersIT,UsersIT` etc etc

... in it. Then we could pull this list into whatever script.